### DISCUSSION TITLE
Python and the ReplAPI-It Packages

### DISCUSSION THREAD

**Discussion by BD103** [ReplAPI-it/ReplAPI.it-NodeJS#56] [Improvements/R&D] [answered]
How is your stance on supporting more than just Node.js for programming languages? I'm specifically good at Python, and could probably contribute more if there was something related through there.

- Is ReplAPI-It Javascript specific? Or is it just a collection of helper packages for Replit?
- Say someone wanted to make a C API, or a Java one, would you allow it?

_Just curious :)_

**Comment by abhay7dev** [comment]
Its just getting data from endpoints which teh developer can then use. It can be done in any language.

**Reply by rayhanadev** [reply]
I think what they mean is can you convert the ReplAPI.it Package (like everything that makes this package unique) into other languages.

**Comment by rayhanadev** [comment, marked as answer]
@BD103 Good question!

ReplAPI.it could certainly be expanded to new frontiers if any developers are willing to come up!

The only problem I see is that the package would need to be the same (or as similar as possible) in each language to be considered ReplAPI.it (otherwise it's just another Replit API package) and any updates in the main (which would be the NodeJS package) would need to be reflected across each package.

Also you and @EpicGamer007 are on the team, you don't need to ask me for permission for anything just make anything you feel is relevant to the package or Replit API it's why I want you two on this team xD.

If it interests you, you may head up a Python transpilation of the package (and @EpicGamer007 same applies to you if you want to do Java) just create a new repository under the team and a new Repl in the the team to get started.

**Reply by rayhanadev** [reply]
@BD103 @EpicGamer007 also if you feel like anyone could help you out with this project [shoot me an email](mailto:rayhanadev@protonmail.com) or DM me on Discord and I'll get back to you on whether I feel like they should be on the team.

**Reply by kokonut27** [reply]
@RayhanADev, heyo, quick question, could you invite me (you don’t have to) for the python version of the package?

**Reply by abhay7dev** [reply]
I am not sure about graphql yet and i still need some work todo with java but i will take a look someday maybe

**Reply by BD103** [reply]
Thanks, @RayhanADev. I'll probably need to understand GraphQL for Python first, but then I'll think about it!

**Reply by rayhanadev** [reply]
> @RayhanADev, heyo, quick question, could you invite me (you don’t have to) for the python version of the package?

@JBYT27 I would ask @BD103 (assuming they want to head up the Python project) :). I don't want to "intrude" on it.

**Reply by rayhanadev** [reply]
> I am not sure about graphql yet and i still need some work todo with java but i will take a look someday maybe

@EpicGamer007 works for me :)

**Reply by kokonut27** [reply]
@RayhanADev, ah, ok. Sure thing. :)

**Reply by rayhanadev** [reply]
> Thanks, @RayhanADev. I'll probably need to understand GraphQL for Python first, but then I'll think about it!

sure! Also you can use the requests library and send it via requests (look at JB and Dark's package to see what I did with them)

**Reply by BD103** [reply]
@RayhanADev The repo is created, I'll start working on it soon!

**Comment by kokonut27** [comment]
Hiya @BD103, 

So I asked Ray already, but I'm asking you because your kinda in-charge of the python package version. Basically, could I work with you on the python package version? If not, you don't have to worry ;)

Thanks!

**Reply by BD103** [reply]
Sure, let me set up a repo

I won't be able to fully work on it yet, but I can outline an API

**Reply by kokonut27** [reply]
Ok, thanks! 

Same for me, I might have to work on smth else. But thanks so much!

**Reply by BD103** [reply]
The repo is created. I'll start working near the end of this week.

**Reply by kokonut27** [reply]
Uh, did you invite me to the repl/repo?
